### DISCUSSION TITLE
best file encoding and line endings for cross-plattform PowerShell modules

### DISCUSSION THREAD

**Discussion by mwallner** [PowerShell/PowerShell#18303] [Q&A]
I am struggling to find a resource that points out the advantages and disadvantages of different encoding and linefeed variants for .ps1 / .psm1 / .psd1 files.

Context: creating PowerShell modules that are supposed to run on Windows/Unix and both Windows PowerShell 5+ as well as PowerShell 7.

Currently I'm leaning towards thinking that UTF8BOM and CRLF is the best option for that matter.

**Comment by rhubarb-geek-nz** [comment]
It depends how the text files are shared between/packaged/delivered to the platforms. For example, if they are held in `git` and then cloned, then the line endings will depend on the system doing the clone as `git` will internally hold the files with just LF.
